src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -677,6 +686,7 @@ public void run() {
                     LOG.debug("Error closing server socket", ie);
                 }
             }
+            if (ce !=null) throw new RuntimeException(ce);
```

Comment:
Okay, I am still unsure if throwing a Runtime exception is the best approach here.